pkg/receive/writer.go

```diff
@@ -28,6 +29,26 @@ type TenantStorage interface {
 	TenantAppendable(string) (Appendable, error)
 }
 
+// Wraps storage.Appender to add validation and logging.
+type ReceiveAppender struct {
```

Comment:
Sorry if my comment wasn't clear, I meant that this appender should always reject samples that are too far in the future.

If the flag is set to a value other than zero, we would wrap the default appender with this one. Otherwise we would just use the storage appender as before.

Reply:
ic, we can do that, but I feel this pattern is more extensible for wrapping prometheus' default appender implementation going forward:
```
app := multiTSDB.appender(tenantID)
if tooFarInFuture != 0 {
  app = ReceiveAppender{ toofarInFuture, app}
} else if <another condition> {
 app = AnotherAppender{ flag, app}
}
```
vs
```
app :=  ReceiveAppender{
  tooFarInFuture: tooFarInFutureFlag,
  // other flags
  app: multiTSDB.appender(tenantID)
}

// add behavior insider `ReceiveAppender`
```

Reply:
The appender wrapper would only be responsible for rejecting samples in the future and other appender wrappers will apply different limits. So it would rather be:

```
app := multiTSDB.appender(tenantID)
if tooFarInFuture != 0 {
  app = AppenderWithFutureLimit(toofarInFuture, app}
} 
if <another condition> {
 app = AnotherAppender{flag, app}
} 
if <third condition> {
  app = ThirdAppender(flag, appender)
}
```

In any case, this is definitely not blocker but more of a suggestion. We would anyway have one appender for now and we can break it up into multiple if it the logic gets more complicated in the future.